File: typescript/tests/integration/session-delete-integration.test.ts
import { AgentBay, Session, CreateSessionParams, ContextSync } from "../../src";
import { getTestApiKey } from "../utils/test-helpers";
import { log } from "../../src/utils/logger";

// Helper to conditionally run tests
const itif = it;

// Helper to generate a unique ID
function generateUniqueId(): string {
  return Math.random().toString(36).substring(2, 10);
}

describe("SessionDeleteIntegration", () => {
  let agentBay: AgentBay;

  beforeAll(() => {
    const apiKey = getTestApiKey();
    agentBay = new AgentBay({ apiKey });
  });

  itif("should delete session without parameters", async () => {
    // Create a new session
    log("Creating a new session for delete testing...");
    const createResponse = await agentBay.create();
    expect(createResponse.success).toBe(true);

    const session = createResponse.session;
    log(`Session created with ID: ${session.sessionId}`);

    // Delete session without parameters
    log("Deleting session without parameters...");
    const deleteResult = await session.delete();
    expect(deleteResult.success).toBe(true);
    log(`Session deleted (RequestId: ${deleteResult.requestId || "undefined"})`);

    // Verify the session was deleted
    const listResult = await agentBay.listByLabels();
    expect(listResult.success).toBe(true);
    
    // Check that our session is not in the list
    const sessions = listResult.data || [];
    for (const s of sessions) {
      expect(s.sessionId).not.toBe(session.sessionId);
    }
  });

  itif("should delete session with syncContext=true", async () => {
    try {
      // Create a context
      const contextName = `test-context-${generateUniqueId()}`;
      log(`Creating context: ${contextName}...`);
      const contextResult = await agentBay.context.create(contextName);
      expect(contextResult.contextId).toBeDefined();
      expect(contextResult.contextId.length).toBeGreaterThan(0);
      
      const contextId = contextResult.contextId;
      log(`Context created with ID: ${contextId}`);

      // Create persistence configuration
      const persistenceData = [
        {
          contextId: contextId,
          path: "/home/wuying/test"
        } as ContextSync
      ];

      // Create a session with context
      const params: CreateSessionParams = {
        imageId: "linux_latest",
        contextSync: persistenceData
      };

      log("Creating a session with context...");
      const createResponse = await agentBay.create(params);
      expect(createResponse.success).toBe(true);

      const session = createResponse.session;
      log(`Session created with ID: ${session.sessionId}`);

      // Create a test file in the session
      try {
        const testCmd = "echo 'test file content' > /home/wuying/test/testfile.txt";
        const cmdResult = await session.command.executeCommand(testCmd);
        log(`Created test file: ${cmdResult}`);
      } catch (error) {
        log(`Warning: Failed to create test file: ${error}`);
      }

      // Delete session with syncContext=true
      log("Deleting session with syncContext=true...");
      const deleteResult = await session.delete(true);
      expect(deleteResult.success).toBe(true);
      log(`Session deleted with syncContext=true (RequestId: ${deleteResult.requestId || "undefined"})`);

      // Verify the session was deleted
      const listResult = await agentBay.listByLabels();
      expect(listResult.success).toBe(true);
      
      // Check that our session is not in the list
      const sessions = listResult.data || [];
      for (const s of sessions) {
        expect(s.sessionId).not.toBe(session.sessionId);
      }

      // Clean up context
      try {
        const getContextResult = await agentBay.context.get(contextName);
        if (getContextResult.context) {
          const deleteContextResult = await agentBay.context.delete(getContextResult.context);
          log(`Context ${contextId} deleted (Success: ${deleteContextResult.success})`);
        }
      } catch (error) {
        log(`Warning: Failed to clean up context: ${error}`);
      }
    } catch (error) {
      fail(`Test failed with error: ${error}`);
    }
  });

  itif("should delete session using AgentBay.delete with syncContext=true", async () => {
    try {
      // Create a context
      const contextName = `test-context-${generateUniqueId()}`;
      log(`Creating context: ${contextName}...`);
      const contextResult = await agentBay.context.create(contextName);
      expect(contextResult.contextId).toBeDefined();
      expect(contextResult.contextId.length).toBeGreaterThan(0);
      
      const contextId = contextResult.contextId;
      log(`Context created with ID: ${contextId}`);

      // Create persistence configuration
      const persistenceData = [
        {
          contextId: contextId,
          path: "/home/wuying/test2"
        } as ContextSync
      ];

      // Create a session with context
      const params: CreateSessionParams = {
        imageId: "linux_latest",
        contextSync: persistenceData
      };

      log("Creating a session with context...");
      const createResponse = await agentBay.create(params);
      expect(createResponse.success).toBe(true);

      const session = createResponse.session;
      log(`Session created with ID: ${session.sessionId}`);

      // Create a test file in the session
      try {
        const testCmd = "echo 'test file for agentbay delete' > /home/wuying/test2/testfile2.txt";
        const cmdResult = await session.command.executeCommand(testCmd);
        log(`Created test file: ${cmdResult}`);
      } catch (error) {
        log(`Warning: Failed to create test file: ${error}`);
      }

      // Delete session with AgentBay.delete and syncContext=true
      log("Deleting session with AgentBay.delete and syncContext=true...");
      const deleteResult = await agentBay.delete(session, true);
      expect(deleteResult.success).toBe(true);
      log(`Session deleted with AgentBay.delete and syncContext=true (RequestId: ${deleteResult.requestId || "undefined"})`);

      // Verify the session was deleted
      const listResult = await agentBay.listByLabels();
      expect(listResult.success).toBe(true);
      
      // Check that our session is not in the list
      const sessions = listResult.data || [];
      for (const s of sessions) {
        expect(s.sessionId).not.toBe(session.sessionId);
      }

      // Clean up context
      try {
        const getContextResult = await agentBay.context.get(contextName);
        if (getContextResult.context) {
          const deleteContextResult = await agentBay.context.delete(getContextResult.context);
          log(`Context ${contextId} deleted (Success: ${deleteContextResult.success})`);
        }
      } catch (error) {
        log(`Warning: Failed to clean up context: ${error}`);
      }
    } catch (error) {
      fail(`Test failed with error: ${error}`);
    }
  });
}); 

File: golang/tests/pkg/integration/session_delete_integration_test.go
package integration

import (
	"fmt"
	"os"
	"testing"
	"time"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
)

// TestSessionDeleteWithoutParams integration test for Delete without parameters
func TestSessionDeleteWithoutParams(t *testing.T) {
	// Get API Key
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Fatal("AGENTBAY_API_KEY environment variable is not set")
	}

	// Initialize AgentBay client
	client, err := agentbay.NewAgentBay(apiKey, nil)
	if err != nil {
		t.Fatalf("Failed to initialize AgentBay client: %v", err)
	}

	// Create a session
	fmt.Println("Creating a new session for delete testing...")
	result, err := client.Create(nil)
	if err != nil {
		t.Fatalf("Failed to create session: %v", err)
	}
	session := result.Session
	t.Logf("Session created with ID: %s", session.SessionID)

	// Delete session using default parameters
	fmt.Println("Deleting session without parameters...")
	deleteResult, err := session.Delete()
	if err != nil {
		t.Fatalf("Failed to delete session: %v", err)
	}
	t.Logf("Session deleted (RequestID: %s)", deleteResult.RequestID)

	// Verify session has been deleted
	listResult, err := client.ListByLabels(agentbay.NewListSessionParams())
	if err != nil {
		t.Fatalf("Failed to list sessions: %v", err)
	}

	for _, sessionId := range listResult.SessionIds {
		if sessionId == session.SessionID {
			t.Errorf("Session with ID %s still exists after deletion", session.SessionID)
		}
	}
}

// TestAgentBayDeleteWithSyncContext integration test for AgentBay.Delete with syncContext parameter
func TestAgentBayDeleteWithSyncContext(t *testing.T) {
	// Get API Key
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Fatal("AGENTBAY_API_KEY environment variable is not set")
	}

	// Initialize AgentBay client
	client, err := agentbay.NewAgentBay(apiKey, nil)
	if err != nil {
		t.Fatalf("Failed to initialize AgentBay client: %v", err)
	}

	// Create context
	contextName := "test-context-" + time.Now().Format("20060102150405")
	fmt.Println("Creating a new context...")
	createResult, err := client.Context.Create(contextName)
	if err != nil {
		t.Fatalf("Failed to create context: %v", err)
	}
	contextID := createResult.ContextID
	t.Logf("Context created with ID: %s", contextID)

	// Create persistence configuration
	persistenceData := []*agentbay.ContextSync{
		{
			ContextID: contextID,
			Path:      "/home/wuying/test",
		},
	}

	// Create session with context
	params := agentbay.NewCreateSessionParams()
	params.ImageId = "linux_latest"
	params.ContextSync = persistenceData

	fmt.Println("Creating a new session with context...")
	result, err := client.Create(params)
	if err != nil {
		t.Fatalf("Failed to create session: %v", err)
	}
	session := result.Session
	t.Logf("Session created with ID: %s", session.SessionID)

	// Create 1GB test file in session
	fmt.Println("Creating a 1GB test file for agentbay delete...")
	testCmd := "dd if=/dev/zero of=/home/wuying/test/testfile2.txt bs=1M count=1024"
	cmdResult, err := session.Command.ExecuteCommand(testCmd)
	if err != nil {
		t.Logf("Warning: Failed to create 1GB test file: %v", err)
	} else {
		t.Logf("Created 1GB test file: %s", cmdResult)
	}

	// Delete session using client.Delete with syncContext=true
	fmt.Println("Deleting session with AgentBay.Delete and syncContext=true...")
	deleteResult, err := client.Delete(session, true)
	if err != nil {
		t.Fatalf("Failed to delete session: %v", err)
	}
	t.Logf("Session deleted with client.Delete and syncContext=true (RequestID: %s)", deleteResult.RequestID)

	// Verify session has been deleted
	listResult, err := client.ListByLabels(agentbay.NewListSessionParams())
	if err != nil {
		t.Fatalf("Failed to list sessions: %v", err)
	}

	for _, sessionId := range listResult.SessionIds {
		if sessionId == session.SessionID {
			t.Errorf("Session with ID %s still exists after deletion", session.SessionID)
		}
	}

	// Clean up context
	// Create Context object for deletion
	getResult, err := client.Context.Get(contextName, false)
	if err == nil && getResult != nil && getResult.Context != nil {
		_, err = client.Context.Delete(getResult.Context)
		if err != nil {
			t.Logf("Warning: Failed to delete context: %v", err)
		} else {
			t.Logf("Context %s deleted", contextID)
		}
	}
}
